src/coreclr/jit/emitriscv64.cpp

```diff
@@ -2748,48 +2748,50 @@ ssize_t emitter::emitOutputInstrJumpDistance(const BYTE* src, const insGroup* ig
     return distVal;
 }
 
-static constexpr size_t NBitMask(uint8_t bits)
+template <uint8_t Bits>
+static inline constexpr typename std::conditional<(Bits > 32), size_t, unsigned>::type NBitMask() noexcept
 {
-    return (static_cast<size_t>(1) << bits) - 1;
+    return static_cast<typename std::conditional<(Bits > 32), size_t, unsigned>::type>(
+        (static_cast<size_t>(1) << Bits) - 1);
```

Comment:
Are there any uses that need a 64 bit mask? IMO just making it unconditionally return an `unsigned` would be significantly more readable than this.

Reply:
I was thinking about future refactors where some longer masks could be useful, but you're right, for now the 32-bit version is sufficient enough